### PR TITLE
Improve perf of cli

### PR DESCRIPTION
**Pull request by Coobaha**
These changes significantly improve CLI perf

**Before**

image

**After**

image

Note that the initial 5s is a fixed cost; the difference is even bigger for multiple files. For complex files, it speedups 16x (5s vs 300ms)

Hopefully, it doesn't produce any extra diff. At least my local qa and `jest` passes

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -57,39 +80,36 @@ function main(filePaths, listDifferent, ignorePunctuation) {
...
-
-      if (/^tsconfig.*\.json$/gi.test(path.basename(filePath))) {
+      const readProject = new Project({
+        tsConfigFilePath, manipulationSettings, compilerOptions: readCompilerOptions
+      });
+      const project = new Project({
+        manipulationSettings,
+        skipAddingFilesFromTsConfig: true,
+        skipFileDependencyResolution: true,
+        compilerOptions: readCompilerOptions
```
Comment (nvh) [inline cli.js:96]: If we're using the `readCompilerOptions` for both projects, why are they not just called `compilerOptions`?

File: cli.js [hunk 2]
```diff
@@ -135,9 +157,36 @@ function main(filePaths, listDifferent, ignorePunctuation) {
...
+        includeInlayParameterNameHintsWhenArgumentMatchesName: false,
+        includeInlayFunctionParameterTypeHints: false,
+        includeInlayVariableTypeHints: false,
+        includeInlayVariableTypeHintsWhenTypeMatchesName: false,
+        includeInlayPropertyDeclarationTypeHints: false,
+        includeInlayFunctionLikeReturnTypeHints: false,
+        includeInlayEnumMemberValueHints: false,
+        autoImportFileExcludePatterns: ["**/*"],
+        interactiveInlayHints: false,
+        quotePreference: "double"
```
Comment (nvh) [inline cli.js:188]: How does this differ from the defaults? seems like a lot of flags 😅

File: package.json [hunk 3]
```diff
@@ -24,6 +24,6 @@
   "scripts": {
     "all-checks": "yarn test && yarn run typecheck",
     "test": "jest",
-    "typecheck": "tsc --checkJs --allowJs --noEmit --strict cli.js"
+    "typecheck": "tsc --checkJs --allowJs --noEmit --skipLibCheck --strict cli.js"
```
Comment (nvh) [inline package.json:27]: What does this do?

**Review by nvh — APPROVED** [review]

File: cli.js (hunk 2 above)
Comment (Coobaha) [reply cli.js:188]: :D I think we can indeed remove them, no big difference

File: package.json (hunk 3 above)
Comment (Coobaha) [reply package.json:27]: it skips errors from node_modules

**Pull request closed without merging**